internal/util/vault.go

```diff
@@ -65,73 +62,110 @@ Example JSON structure in the KMS config is,
-	if !ok || kms.VaultRole == "" {
-		kms.VaultRole = vaultDefaultRole
+	vaultConfig[vault.AuthMountPath], err = detectAuthMountPath(vaultAuthPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set %s in Vault config: %w", vault.AuthMountPath, err)
 	}
-	kms.VaultNamespace, ok = config["vaultNamespace"]
-	if !ok || kms.VaultNamespace == "" {
-		kms.VaultNamespace = vaultDefaultNamespace
+
+	vaultRole, ok := config["vaultRole"]
+	if !ok || vaultRole == "" {
+		vaultRole = vaultDefaultRole
 	}
-	kms.VaultPassphraseRoot, ok = config["vaultPassphraseRoot"]
-	if !ok || kms.VaultPassphraseRoot == "" {
-		kms.VaultPassphraseRoot = vaultDefaultPassphraseRoot
+	vaultConfig[vault.AuthKubernetesRole] = vaultRole
+
+	vaultNamespace, ok := config["vaultNamespace"]
+	if !ok || vaultNamespace == "" {
+		vaultNamespace = vaultDefaultNamespace
 	}
-	kms.VaultPassphrasePath, ok = config["vaultPassphrasePath"]
-	if !ok || kms.VaultPassphrasePath == "" {
-		kms.VaultPassphrasePath = vaultDefaultPassphrasePath
+	vaultConfig[api.EnvVaultNamespace] = vaultNamespace
+	keyContext[loss.KeyVaultNamespace] = vaultNamespace
+
+	// vault.VaultBackendPathKey is "secret/" by default, use vaultPassphraseRoot if configured
+	vaultPassphraseRoot, ok := config["vaultPassphraseRoot"]
+	if ok && vaultPassphraseRoot != "" {
+		// the old example did have "/v1/secret/", convert that format
+		if strings.HasPrefix(vaultPassphraseRoot, "/v1/") {
+			vaultConfig[vault.VaultBackendPathKey] = strings.TrimPrefix(vaultPassphraseRoot, "/v1/")
+		} else {
+			vaultConfig[vault.VaultBackendPathKey] = vaultPassphraseRoot
+		}
 	}
-	kms.VaultCAVerify = true
+
+	kms.vaultPassphrasePath, ok = config["vaultPassphrasePath"]
+	if !ok || kms.vaultPassphrasePath == "" {
+		kms.vaultPassphrasePath = vaultDefaultPassphrasePath
+	}
+
 	verifyCA, ok := config["vaultCAVerify"]
 	if ok {
-		kms.VaultCAVerify, err = strconv.ParseBool(verifyCA)
+		var vaultCAVerify bool
+		vaultCAVerify, err = strconv.ParseBool(verifyCA)
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse 'vaultCAVerify' for vault <%s> kms config: %s",
 				kmsID, err)
 		}
+		vaultConfig[api.EnvVaultInsecure] = !vaultCAVerify
 	}
+
 	vaultCAFromSecret, ok := config["vaultCAFromSecret"]
 	if ok && vaultCAFromSecret != "" {
 		caPEM, ok := secrets[vaultCAFromSecret]
 		if !ok {
 			return nil, fmt.Errorf("missing vault CA in secret %s", vaultCAFromSecret)
 		}
-		roots := x509.NewCertPool()
-		ok = roots.AppendCertsFromPEM([]byte(caPEM))
-		if !ok {
-			return nil, fmt.Errorf("failed loading CA bundle for vault from secret %s",
-				vaultCAFromSecret)
+		vaultConfig[api.EnvVaultCACert], err = createTempFile("vault-ca-cert", []byte(caPEM))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create temporary file for Vault CA: %w", err)
 		}
-		kms.vaultCA = roots
+		// TODO: delete f.Name() when VaultKMS is destroyed
+	}
+
+	// FIXME: vault.AuthKubernetesTokenPath is not enough? EnvVaultToken needs to be set?
```

Comment:
FixMe is still needed?

Reply:
Well, 'need' maybe not, but it feels like the configuration is done twice, one option might be sufficient. Will keep it as is, it works for now (any likely needs updating when we want to support multiple Vault tokens instead of the ServiceAccount).